File: Python/8py 3/remove_every_other.py
"""
|--------------------------------------------------------------------------
| Removing Elements
|--------------------------------------------------------------------------
|
"""

def remove_every_other(my_list):
   res = []
   for i in range(len(my_list)):
      if i % 2 == 0:
         res.append(my_list[i])
   return res


def remove_every_other(my_list):
   return my_list[::2]


def remove_every_other(my_list):
   return [v for c,v in enumerate(my_list) if not c%2]


def remove_every_other(my_list):
   return [my_list[i] for i in range(0,len(my_list),2)]


print(remove_every_other(['Hello', 'Goodbye', 'Hello Again']), ['Hello', 'Hello Again'])
print(remove_every_other([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]), [1, 3, 5, 7, 9])

File: JavaScript/8js 2/removeEveryOther.js
/*
|--------------------------------------------------------------------------
| Removing Elements
|--------------------------------------------------------------------------
|
*/

function removeEveryOther(arr) {
   let res = [];
   for (let i = 0; i < arr.length; i++) {
      if (i % 2 == 0) {
         res.push(arr[i]);
      }
   }
   return res;
}


const removeEveryOther = arr => arr.filter((item, i) => i % 2 == 0);


function removeEveryOther(arr) {
   for (var i = 1; i < arr.length; i++) {
      arr.splice(i, 1);
   }
   return arr;
}


function removeEveryOther(arr) {
   return arr.filter((v, i) => !(i & 1));
}


console.log(removeEveryOther(['Hello', 'Goodbye', 'Hello Again']), ['Hello', 'Hello Again']);
console.log(removeEveryOther([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]), [1, 3, 5, 7, 9]);